query-sql/src/lib.rs

```diff
@@ -143,7 +148,7 @@ pub struct Join {
 pub enum TableOrSubquery {
     Table(SourceAlias),
     Union(Vec<SelectQuery>, TableAlias),
-    // TODO: Subquery.
+    Subquery(Rc<SelectQuery>),
```

Comment:
Oh, you have `Sized` problems. `Box` is your friend; `Rc` is specifically for reference counting. `Box` just means "put this on the heap and let the `Box` own it".

You'll see that `Vec` above accidentally serves the same purpose.